### PR TITLE
Implemented a root component for the UI on which the rest of the UI can be built. Also implemented tests for said component.

### PR DESCRIPTION
**Pull request by Szsaunders**

### CODE REVIEW

File: projects/revaturexyz/profile/src/lib/profile.component.spec.ts [hunk 1]
```diff
@@ -1,25 +1,29 @@
-import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+// import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (pgrimes88) [inline projects/revaturexyz/profile/src/lib/profile.component.spec.ts]: Everything in this file is commented out

File: projects/revaturexyz/profile/src/lib/components/filter-panel/filter-panel.component.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+  housingSituation = null;
+  // Check box booleans
+  vacantRoomsOnly: boolean = null;
+  sortByMostVacancies: boolean = null;
+  // unhousedUsersOnly: boolean = true;
+  assignedUsers: boolean = null;
+  hasBedAvailable: boolean = null;
+
+  // Output object
```
Comment (pgrimes88) [inline projects/revaturexyz/profile/src/lib/components/filter-panel/filter-panel.component.ts]: lone comment

**Comment by escalonn** [comment]
name of PR should be more specific and there should be a functional description which someone could use to decide whether the code implemented fulfilled the requirements

File: projects/revaturexyz/profile/src/lib/profile.module.ts [hunk 3]
```diff
@@ -1,10 +1,64 @@
...
+    MatIconModule,
+    MatInputModule,
+    MatOptionModule,
+    MatPaginatorModule,
+    MatRadioModule,
+    MatSelectModule,
+    MatSidenavModule,
+    MatSortModule,
+    MatTableModule,
+    MatTabsModule
```
Comment (escalonn) [inline projects/revaturexyz/profile/src/lib/profile.module.ts]: are we really using all these modules?

File: projects/revaturexyz/profile/src/lib/profile.module.ts [hunk 4]
```diff
@@ -1,10 +1,64 @@
...
+  ],
+  providers: [
   ],
-  exports: [ProfileComponent]
+  exports: [
+    FilterPanelComponent,
+    ProfileComponent
+  ]
```
Comment (escalonn) [inline projects/revaturexyz/profile/src/lib/profile.module.ts]: i don't think we need to export anything besides ProfileComponent

File: projects/revaturexyz/profile/src/lib/components/filter-panel/filter-panel.component.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * Provides a UI that allows the user to filter and sort
+ * the data shown in other components.
+ */
+@Component({
+  selector: 'lib-side-bar',
+  templateUrl: './filter-panel.component.html',
+  styleUrls: ['./filter-panel.component.css']
+})
+export class FilterPanelComponent implements OnInit {
```
Comment (escalonn) [inline projects/revaturexyz/profile/src/lib/components/filter-panel/filter-panel.component.ts]: - filter panel is not a name that makes sense for what is essentially a navbar
- do you really want to make it be hidden by default so the user has to press a button, even on desktop-size screens? is an expando-panel the right tool for this job?

**Comment by Szsaunders** [comment]
A minute after I saw you comment on the PR name, I realized that you were going to correct me about keeping the 'filter-panel' name. I'll count that as self-improvement.

I'm not sure what you mean by the button and the expando-panel? If you're referring to the side-bar on the original version of the filter-panel, I removed that during the streamlining process.

File: projects/revaturexyz/profile/src/lib/profile.module.ts (hunk 3 above)
Comment (escalonn) [reply projects/revaturexyz/profile/src/lib/profile.module.ts]: this comment still not addressed

Comment (escalonn) [reply projects/revaturexyz/profile/src/lib/profile.module.ts]: this comment still not addressed

File: projects/revaturexyz/profile/src/lib/components/tabs/tabs.component.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import { Component, OnInit } from '@angular/core';
+
+// The filter panel of the housing Profile front-end.
+// Provides a UI that allows the user to filter and sort
+// the data shown in other components.
```
Comment (escalonn) [inline projects/revaturexyz/profile/src/lib/components/tabs/tabs.component.ts]: this comment is not correct

File: projects/revaturexyz/profile/src/lib/components/tabs/tabs.component.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+@Component({
+  selector: 'lib-tabs',
+  templateUrl: './tabs.component.html',
+  styleUrls: ['./tabs.component.css']
+})
+export class TabsComponent implements OnInit {
+
+  batchId: string;
+  location: string;
```
Comment (escalonn) [inline projects/revaturexyz/profile/src/lib/components/tabs/tabs.component.ts]: i dont think either of these two properties make sense for this component

File: projects/revaturexyz/profile/src/lib/components/tabs/tabs.component.spec.ts [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+describe('TabsComponent', () => {
+  let component: TabsComponent;
+  let fixture: ComponentFixture<TabsComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        FormsModule,
+        HttpClientModule,
+        CommonModule,
```
Comment (escalonn) [inline projects/revaturexyz/profile/src/lib/components/tabs/tabs.component.spec.ts]: you definitely dont need httpclientmodule, and pretty sure you dont need any of the others either.

File: projects/revaturexyz/profile/src/lib/components/tabs/tabs.component.ts (hunk 6 above)
Comment (escalonn) [reply projects/revaturexyz/profile/src/lib/components/tabs/tabs.component.ts]: this comment still not addressed.

**Pull request merged**